### ISSUE TITLE
Box shadow doesn't have flexible features.

### ISSUE THREAD

**Issue by mohith7548** [flutter/flutter#20775]
I love flutter. I have been using it for a year. I'm working on a client project. And the demand is We need Box-shadow for only 3 sides. How can I achieve this? If it would have been like `BoxShadow.only` as in the case of `EdgeInsted.only`, that would be much better. So that as a Developer, I can't say "No" to my Client. And that's what Flutter is For. I don't know where to suggest this. So I'm reporting here.
![37932941_1016917551815921_3708467126361653248_n](https://user-images.githubusercontent.com/24828445/44294926-bb04b780-a2bc-11e8-9f46-c24b45d734c4.png)

**Comment by albrnick** [comment]
I have a similar case, but only want on 2 sides, opposite each other. So the offset wouldn't work..

**Comment by pepegich** [comment]
@mohith7548 @albrnick I added padding property in PR https://github.com/flutter/flutter/pull/28948

**Comment by dnfield** [comment]
I don't think this is a box shadow anymore.  It's certainly an interesting effect, and it shares characteristics with a box shadow, but a shadow implies a light source.  What kind of light source would create a shadow only on two opposing sides?  Is it that we have multiple light sources in this case?  How do we handle it on a platform where shadows are composited so that the camera can actually be moved and the lightsource be tinkered with?

I think it would make sense to make some kind of new BoxDecoration that is _like_ a BoxShadow but customizable in this way - but I dont' think we should be updating BoxShadow for this.

**Comment by albrnick** [comment]
Oh, interesting. I guess I never thought about the name, and what it implied.  Though, but default, I would think if it were from a light source, you should only have 2 sides at any on time?  (And not the default 4?)

But yeah, certainly a different name is fine. I'm just looking for the capability of selecting where the shadow shows up.. :)

**Comment by mayurdhurpate** [comment]
As a (horrible) hack, I created a small rectangular strip of same color and positioned it on the Shadow edge of main Container to hide it. Worked well for me. But of course I agree, the capability of selecting where the shadow shows up would be a great addition.

**Comment by nicoroy2561** [comment]
While it may not make much sense from a "what a shadow actually is/is meant to be" perspective, coming from web design you'll have use cases for all of these things (and, they are fairly easy to make in CSS). So having the same features in Flutter would be truly awesome.
Besides, I'm fine with having a separate BoxShadow for it, as long as there's a way and it's documented somewhere.

**Comment by seansabe** [comment]
There's a issue with BoxShadow where no shadow is displaying in the right and left side of a triangle drawn with a Container.

![Screenshot_1585773064](https://user-images.githubusercontent.com/30910240/78184155-a55eea00-742e-11ea-8ee2-471f79de430e.png)

As you can see in the image, the shadow just extends to the top and bottom corner. There should be an option that allows us to set shadow to all borders or specific borders; something like .all() .only() .symetric() .fromLTRB()
